lib/CfTemplateGenerators/S3.js

```diff
@@ -0,0 +1,18 @@
+function replaceS3BucketFunctionWithAlias(bucket, functionAlias, functionName) {
+  const lambdaConfigurations = bucket.Properties.NotificationConfiguration.LambdaConfigurations;
+  const newConfigurations = lambdaConfigurations.map((configuration) => {
+    const functionObject = configuration.Function;
+    const thisFunctionName = (functionObject['Fn::GetAtt'] || [])[0];
+    if (thisFunctionName !== functionName) return configuration;
+    return Object.assign({}, configuration, { Function: { Ref: functionAlias } });
+  });
+  const newNotificationCfonfiguration = Object.assign({}, bucket.Properties.NotificationConfiguration, { LambdaConfigurations: newConfigurations });
```

Comment:
The last three lines could be safely replaced by `_.set`, which first clones the object and then modifies the property (see [here](https://github.com/davidgf/serverless-plugin-canary-deployments/blob/master/lib/CfTemplateGenerators/Lambda.js#L33) for example)